lib/reek/smells/feature_envy.rb

```diff
@@ -42,6 +42,7 @@ def self.smell_category
       # @return [Array<SmellWarning>]
       #
       def examine_context(method_ctx)
+        return [] unless method_ctx.exp.depends_on_instance?
```

Comment:
Can we abstract the guard away into a speaking method so that we make this idea:

>> Hence this alternative: FeatureEnvy is only triggered if there are some references to self, and UtilityFunction is triggered if there are no references to self.

clear to everybody including ourselves in a couple of months?
How about

```Ruby
return [] if no_references_to_self?(method_ctx.exp)
```

Additionally I'd appreciate if we could put 

>> FeatureEnvy is only triggered if there are some references to self, and UtilityFunction is triggered if there are no references to self.

as a introductory comment at the top of the file.